### PR TITLE
[FR] Update sdk to service v2.0

### LINKED ISSUE #12563: Update service version
When the service releases their GA version, and we consume it, we need to update our docs/samples to match service version

### PR DESCRIPTION
**Pull request by maririos**
Fixes: https://github.com/Azure/azure-sdk-for-net/issues/12563

### CODE REVIEW

**Review by kristapratico — APPROVED** [review]
LGTM

File: sdk/formrecognizer/Azure.AI.FormRecognizer/src/FormRecognizerClientOptions.cs [hunk 1]
```diff
@@ -31,10 +31,10 @@ public FormRecognizerClientOptions(ServiceVersion version = LatestVersion)
         public enum ServiceVersion
         {
             /// <summary>
-            /// The V1 of the template service.
+            /// The V2.0
```
Comment (kinelski) [inline sdk/formrecognizer/Azure.AI.FormRecognizer/src/FormRecognizerClientOptions.cs:34]: `The V2.0 of the template service.`?

File: sdk/formrecognizer/Azure.AI.FormRecognizer/CHANGELOG.md [hunk 2]
```diff
@@ -4,6 +4,7 @@
 
 ### Breaking changes
 
+- Now targets the service's v2.0 API, instead of the v2.0-preview.1 API.
```
Comment (kinelski) [inline sdk/formrecognizer/Azure.AI.FormRecognizer/CHANGELOG.md:7]: The lack of subject sounds a bit weird to me. Maybe:
```
The library now targets the service's...
```
?

**Review by kinelski — APPROVED** [review]
LGTM. Left some nitpicking comments.

**Comment by kinelski** [comment]
By the way, we didn't need to run the code generator?

File: sdk/formrecognizer/Azure.AI.FormRecognizer/src/FormRecognizerClientOptions.cs (hunk 1 above)
Comment (maririos) [reply sdk/formrecognizer/Azure.AI.FormRecognizer/src/FormRecognizerClientOptions.cs:34]: hmm not sure... for me it sounds like template service comes from the template library we have in our repo ... so decided to remove

**Comment by maririos** [comment]
> By the way, we didn't need to run the code generator?

We did. The [`ServiceRestClient`](https://github.com/Azure/azure-sdk-for-net/pull/14026/files#diff-cf2b0f719a09490471ebcb46aa99f1f3R50) was the only thing that changed as the only update in the swagger was to point to `v2.0`

File: sdk/formrecognizer/Azure.AI.FormRecognizer/src/FormRecognizerClientOptions.cs (hunk 1 above)
Comment (kinelski) [reply sdk/formrecognizer/Azure.AI.FormRecognizer/src/FormRecognizerClientOptions.cs:34]: I see. Maybe renaming:
```
The V2.0 of the service?
```

Regardless of our decision, we need to update the `ServiceVersion` description in the same way. Currently it is:
```
The template service version.
```

If you want to keep it as `The V2.0`, just don't forget to add the period `.` at the end.

Comment (maririos) [reply sdk/formrecognizer/Azure.AI.FormRecognizer/src/FormRecognizerClientOptions.cs:34]: :( merged too fast.
I will add this in my next PR.
Thank you!

**Pull request merged**